atlassian_jwt_auth/contrib/django/decorators.py

```diff
@@ -31,41 +27,18 @@ def requires_asap_wrapper(request, *args, **kwargs):
             if isinstance(auth_header, six.string_types):
                 auth_header = auth_header.encode(encoding='iso-8859-1')
             auth = auth_header.split(b' ')
-            if not auth or len(auth) != 2:
-                return HttpResponse('Unauthorized', status=401)
-
-            message = None
-            exception = None
-            try:
-                asap_claims = parse_jwt(verifier, auth[1])
-                verify_issuers(asap_claims, issuers)
-                request.asap_claims = asap_claims
+            result = _requires_asap(
+                verifier=verifier,
+                auth=auth,
+                parse_jwt_func=parse_jwt,
+                response_class=HttpResponse,
+                asap_claim_holder=request,
+                verify_issuers_func=verify_issuers,
+                issuers=issuers,
+            )
+            if result is None:
                 return func(request, *args, **kwargs)
-            except RequestException as e:
-                # Error communicating to get key
-                message = 'Unauthorized: Communications error retrieving key'
-                exception = e
-            except PrivateKeyRetrieverException as e:
-                # Error parsing or getting private key
-                message = 'Unauthorized: Unable to retrieve private key'
-                exception = e
-            except PublicKeyRetrieverException as e:
-                # Error parsing or getting public key
-                message = 'Unauthorized: Unable to retrieve public key'
-                exception = e
-            except InvalidIssuerError as e:
-                message = 'Unauthorized: Invalid token issuer'
-                exception = e
-            except InvalidTokenError as e:
-                # Something went wrong with decoding the JWT
-                message = 'Unauthorized: Invalid token'
-                exception = e
-            if message is not None:
-                logger = logging.getLogger(__name__)
-                logger.error(message,
-                             extra={'original_message': str(exception)})
-
-                return HttpResponse(message, status=401)
+            return result
```

Comment:
Nit: this seems a bit awkward; maybe `s/result/err_response/`?

Alternatively, since we're essentially parameterizing the decorator logic with a helper function, we could change the abstracted function to be a decorator that calls the inner function itself and then have something along the lines of `requires_asap = partial(_requires_asap, **{all the details})` here.

Reply:
Yeah probably `err_response`.